README.md
# ublab_bot
This is an Eggdrop customized bot for #ublab on Freenode. Its written in C and Tcl.

# Building the Bot

cd ublab_bot
run: ./configure && make

Once these finishes, you can then run this command

./eggdrop -m ublab.conf

Hurray!!!! You are ready to start using the bot.

# Customizing the Bot

To customize this bot, just open the ublab.conf file in the root of ublab_bot directory and edit the information in the file then you are done. Follow the steps above then you can use the bot in your channel :)

# Other Docs

If you are have used this bot and understand how it works and want to add the documentation, you are free to do so to help others understand how the bot works. :)


ChangeLog
2011-10-25 10:45  thommey

	* THANKS, doc/Changes1.6, src/patch.h, src/mod/server.mod/server.c:
	  Added missing read trace flag to the nick-len variable trace
	  after being unset to make the trace be removed properly on
	  unload.

2011-10-02 14:51  pseudo

	* doc/: ABOUT, COMPILE-GUIDE, Changes1.6, html/about.html,
	  html/compiling.html: Updated instructions for compilation under
	  Cygwin. Updated space requirements a bit.

2011-10-02 14:11  pseudo

	* doc/Changes1.6, src/patch.h, src/mod/share.mod/share.c: Fixed an
	  ancient bug resulting in sending uninitialized strings when
	  sharing bot addresses.

2011-10-02 12:25  pseudo

	* THANKS, doc/Changes1.6, src/patch.h,
	  src/mod/channels.mod/cmdschan.c: Altered the permission check on
	  the pls/mns ban/exempt/invite commands from USER_MASTER to
	  USER_OP to allow global operators to set global bans.

2011-10-02 12:21  pseudo

	* doc/Changes1.6, src/dcc.c, src/patch.h: Made dcc_telnet_new()
	  allow non-latin characters in new handles like other handle
	  validation functions.

2011-10-02 11:34  pseudo

	* aclocal.m4, configure, doc/Changes1.6, src/patch.h: Removed
	  -mwin32 on Windows. It's no longer necessary and causes problems
	  with Cygwin 1.7.

2011-09-23 14:31  thommey

	* THANKS, aclocal.m4, config.h.in, configure, eggdrop.conf,
	  doc/Changes1.6, doc/html/mod-channels.html, src/bg.c,
	  src/chanprog.c, src/main.c, src/mod/Makefile.in,
	  src/mod/channels.mod/help/chaninfo.help,
	  src/mod/channels.mod/help/channels.help,
	  src/mod/compress.mod/configure, src/mod/dns.mod/configure:
	  backport some 1.8 changes

2011-09-19 08:38  thommey

	* doc/: Changes1.6, html/tcl-commands.html: backport 1.8 docfix in
	  the html tcl-commands for bind cron

2011-09-09 15:37  thommey

	* doc/Changes1.6, src/patch.h, src/tcl.c: Fixed a bug in the
	  mainloop for Tcl threads which is now also being used as
	  vwait/update recursion mainloop.

2011-08-15 12:16  thommey

	* THANKS, doc/Changes1.6, src/mem.c: Removed debug output
	  accidentally checked in.

2011-08-13 16:44  pseudo

	* doc/: Changes1.6, html/tcl-commands.html: Fixed a typo in the
	  html documentation.

2011-08-08 17:45  pseudo

	* doc/Changes1.6, help/cmds2.help: Fixed the .match command help.

2011-08-08 16:37  pseudo

	* doc/Changes1.6, src/patch.h, src/mod/dns.mod/dns.c: Fixed a bug
	  in the backported dns-settings patch where dns_change() would
	  call iptostr() with inappropriate argument.

2011-07-31 14:15  thommey

	* doc/Changes1.6, src/patch.h, src/tcl.c, src/tclhash.h: Partially
	  revert changes to vwait/update.

2011-07-20 04:54  thommey

	* config.h.in, configure, doc/Changes1.6: Ran autotools.

2011-07-20 04:54  thommey

	* aclocal.m4, doc/Changes1.6: More AC_LANG_* usage.

2011-07-20 04:50  thommey

	* aclocal.m4, configure.ac, doc/Changes1.6, src/patch.h,
	  src/mod/channels.mod/channels.c: Work around Tcl8.5.10 bug
	  3371644. Use the autoconf macros AC_LANG_PROGRAM/AC_LANG_SOURCE.

2011-07-09 09:07  thommey

	* THANKS, aclocal.m4, config.h.in, configure.ac, eggdrop.conf,
	  doc/COMPILE-GUIDE, doc/Changes1.6, doc/PATCH-HOWTO,
	  doc/TEXT-SUBSTITUTIONS, doc/Versions, doc/tcl-commands.doc,
	  doc/html/app-textsub.html, doc/html/compiling.html,
	  doc/html/patch-howto.html, doc/html/tcl-commands.html,
	  src/Makefile.in, src/dns.c, src/eggdrop.h, src/main.c,
	  src/main.h, src/mem.c, src/misc.c, src/modules.c, src/modules.h,
	  src/patch.h, src/proto.h, src/tcl.c, src/tclegg.h, src/tclhash.c,
	  src/compat/Makefile.in, src/compat/compat.h, src/compat/strdup.c,
	  src/compat/strdup.h, src/md5/Makefile.in, src/mod/module.h,
	  src/mod/assoc.mod/Makefile, src/mod/blowfish.mod/Makefile,
	  src/mod/channels.mod/Makefile, src/mod/compress.mod/Makefile.in,
	  src/mod/compress.mod/configure.ac, src/mod/console.mod/Makefile,
	  src/mod/ctcp.mod/Makefile, src/mod/dns.mod/Makefile.in,
	  src/mod/dns.mod/configure.ac, src/mod/filesys.mod/Makefile,
	  src/mod/irc.mod/Makefile, src/mod/notes.mod/Makefile,
	  src/mod/seen.mod/Makefile, src/mod/server.mod/Makefile,
	  src/mod/share.mod/Makefile, src/mod/transfer.mod/Makefile,
	  src/mod/uptime.mod/Makefile, src/mod/wire.mod/Makefile,
	  src/mod/woobie.mod/Makefile: Backport changes from 1.8. Set
	  eggdrop version to 1.6.21.

2011-04-01 06:22  pseudo

	* doc/Changes1.6, src/patch.h, src/mod/channels.mod/udefchan.c:
	  Fixed getudef() to return intptr_t instead of int to prevent
	  crashes on 64-bit systems.  Backport of the 1.8 patch.

2011-04-01 06:10  pseudo

	* aclocal.m4, config.h.in, configure, doc/Changes1.6,
	  src/eggdrop.h, src/patch.h, src/mod/dns.mod/configure,
	  src/mod/dns.mod/configure.ac: Fixed dns.mod on QNX6/Mac OS
	  X/Solaris. Rewrote dns.mod's autoconf checks to detect the
	  resolver library properly on Darwin/Solaris/QNX6.  Backport of
	  the 1.8 patch.

2011-04-01 05:59  pseudo

	* eggdrop.conf, doc/Changes1.6, doc/html/mod-dns.html,
	  doc/settings/mod.dns, src/patch.h, src/mod/dns.mod/coredns.c,
	  src/mod/dns.mod/dns.c: Replaced most dns.mod preprocessor
	  definitions with config variables. Added support for
	  user-specified dns servers and non-standard ports. Backport of
	  the 1.8 patch.

2011-02-13 08:19  simple

	* AUTHORS, CONTENTS, FEATURES, INSTALL, NEWS, README, THANKS,
	  aclocal.m4, config.h.in, configure, configure.ac, doc/ABOUT,
	  doc/BANS, doc/BOTNET, doc/COMPILE-GUIDE, doc/CONTENTS,
	  doc/FIRST-SCRIPT, doc/KNOWN-PROBLEMS, doc/MODULES, doc/PARTYLINE,
	  doc/PATCH-HOWTO, doc/TEXT-SUBSTITUTIONS, doc/TRICKS, doc/USERS,
	  doc/Versions, doc/WEIRD-MESSAGES, doc/tcl-commands.doc,
	  doc/html/about.html, doc/html/app-problems.html,
	  doc/html/app-textsub.html, doc/html/app-tricks.html,
	  doc/html/app-weird.html, doc/html/bans.html,
	  doc/html/botnet.html, doc/html/compiling.html,
	  doc/html/egg-core.html, doc/html/faqs.html,
	  doc/html/features.html, doc/html/index.html,
	  doc/html/install.html, doc/html/mod-assoc.html,
	  doc/html/mod-blowfish.html, doc/html/mod-channels.html,

config.h
/* config.h.  Generated from config.h.in by configure.  */
/* config.h.in.  Generated from configure.ac by autoheader.  */

/*
 * Copyright (C) 1997 Robey Pointer
 * Copyright (C) 1999 - 2011 Eggheads Development Team
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 */

#ifndef _EGG_CONFIG_H
#define _EGG_CONFIG_H

/* Define if building universal (internal helper macro) */
/* #undef AC_APPLE_UNIVERSAL_BUILD */

/* Define if running on Mac OS X with dns.mod. */
/* #undef BIND_8_COMPAT */

/* Define if running on NeXT Step. */
/* #undef BORGCUBES */

/* Define to use Eggdrop's snprintf functions regardless of HAVE_SNPRINTF. */
/* #undef BROKEN_SNPRINTF */

/* Define if running under Cygwin. */
/* #undef CYGWIN_HACKS */

/* Define for context debugging. */
#define DEBUG_CONTEXT 1

/* Define if running on SunOS 4.0. */
/* #undef DLOPEN_1 */

/* Arguments passed to configure */
#define EGG_AC_ARGS " '--with-tcllib=/usr/lib/x86_64-linux-gnu/libtcl8.6.so' '--with-tclinc=/usr/include/tcl/tcl.h'"

/* Define the maximum length of handles on the bot. */
/* #undef EGG_HANDLEN */

/* Defines the extension of Eggdrop modules. */
#define EGG_MOD_EXT "so"

/* Defines the current Eggdrop version. */
#define EGG_VERSION 10621

/* Define to 1 if you have the <arpa/inet.h> header file. */
#define HAVE_ARPA_INET_H 1

/* Define to 1 if you have the `clock' function. */
#define HAVE_CLOCK 1

/* Define to 1 if you have the declaration of `tzname', and to 0 if you don't.
   */
/* #undef HAVE_DECL_TZNAME */

/* Define to 1 if you have the <dirent.h> header file, and it defines `DIR'.
   */
#define HAVE_DIRENT_H 1

/* Define to 1 if you have the <dlfcn.h> header file. */
#define HAVE_DLFCN_H 1

/* Define to 1 if you have the `dlopen' function. */
#define HAVE_DLOPEN 1

/* Define to 1 if you have the <dl.h> header file. */
/* #undef HAVE_DL_H */

/* Define to 1 if you don't have `vprintf' but do have `_doprnt.' */
/* #undef HAVE_DOPRNT */

/* Define to 1 if you have the `dprintf' function. */
#define HAVE_DPRINTF 1

/* Define to 1 if you have the <fcntl.h> header file. */
#define HAVE_FCNTL_H 1

/* Define to 1 if you have the `fsync' function. */
#define HAVE_FSYNC 1

/* Define to 1 if you have the `getdtablesize' function. */
#define HAVE_GETDTABLESIZE 1

/* Define to 1 if you have the `getpagesize' function. */
#define HAVE_GETPAGESIZE 1

/* Define to 1 if you have the `getrusage' function. */
#define HAVE_GETRUSAGE 1

/* Define to 1 if you have the `inet_aton' function. */
#define HAVE_INET_ATON 1

/* Define to 1 if the system has the type `intptr_t'. */
#define HAVE_INTPTR_T 1

/* Define to 1 if you have the <inttypes.h> header file. */
#define HAVE_INTTYPES_H 1

/* Define to 1 if you have the `isascii' function. */
#define HAVE_ISASCII 1

/* Define to 1 if you have the `dl' library (-ldl). */
#define HAVE_LIBDL 1

/* Define to 1 if you have the `dld' library (-ldld). */
/* #undef HAVE_LIBDLD */

/* Define to 1 if you have the `dns' library (-ldns). */
/* #undef HAVE_LIBDNS */

/* Define to 1 if you have the `nsl' library (-lnsl). */
#define HAVE_LIBNSL 1

/* Define to 1 if you have the `socket' library (-lsocket). */
/* #undef HAVE_LIBSOCKET */

/* Define to 1 if you have the <limits.h> header file. */
#define HAVE_LIMITS_H 1

/* Define to 1 if you have the `load' function. */
/* #undef HAVE_LOAD */

/* Define to 1 if you have the <loader.h> header file. */
/* #undef HAVE_LOADER_H */

/* Define to 1 if you have the <locale.h> header file. */
#define HAVE_LOCALE_H 1

/* Define to 1 if you have the `lrand48' function. */
#define HAVE_LRAND48 1

/* Define to 1 if you have the <mach-o/dyld.h> header file. */
/* #undef HAVE_MACH_O_DYLD_H */

/* Define to 1 if you have the <mach-o/rld.h> header file. */
/* #undef HAVE_MACH_O_RLD_H */

/* Define to 1 if you have the `mbrlen' function. */
#define HAVE_MBRLEN 1

/* Define to 1 if you have the `memcpy' function. */
#define HAVE_MEMCPY 1

/* Define to 1 if you have the <memory.h> header file. */
#define HAVE_MEMORY_H 1

/* Define to 1 if you have the `memset' function. */
#define HAVE_MEMSET 1

/* Define to 1 if you have a working `mmap' system call. */
#define HAVE_MMAP 1

/* Define to 1 if you have the <ndir.h> header file, and it defines `DIR'. */
/* #undef HAVE_NDIR_H */

/* Define to 1 if you have the <netdb.h> header file. */
#define HAVE_NETDB_H 1

/* Define to 1 if you have the <netinet/in.h> header file. */
#define HAVE_NETINET_IN_H 1

/* Define to 1 if you have the `NSLinkModule' function. */
/* #undef HAVE_NSLINKMODULE */

/* Define to 1 if you have the `rand' function. */
#define HAVE_RAND 1

/* Define to 1 if you have the `random' function. */
#define HAVE_RANDOM 1

/* Define to 1 if you have the `rename' function. */
#define HAVE_RENAME 1

/* Define to 1 if you have the <rld.h> header file. */
/* #undef HAVE_RLD_H */

/* Define to 1 if you have the `rld_load' function. */
/* #undef HAVE_RLD_LOAD */

/* Define to 1 if you have the `setpgid' function. */
#define HAVE_SETPGID 1

/* Define to 1 if you have the `shl_load' function. */
/* #undef HAVE_SHL_LOAD */

/* Define to 1 if you have the `sigaction' function. */
#define HAVE_SIGACTION 1

/* Define to 1 if you have the `sigemptyset' function. */
#define HAVE_SIGEMPTYSET 1

/* Define to 1 if you have the `snprintf' function. */
#define HAVE_SNPRINTF 1

/* Define to 1 if you have the `socklen_t' type. */
#define HAVE_SOCKLEN_T 1

/* Define to 1 if you have the <stdarg.h> header file. */
#define HAVE_STDARG_H 1

/* Define to 1 if you have the <stddef.h> header file. */
#define HAVE_STDDEF_H 1

/* Define to 1 if you have the <stdint.h> header file. */
#define HAVE_STDINT_H 1

/* Define to 1 if you have the <stdio.h> header file. */
#define HAVE_STDIO_H 1

/* Define to 1 if you have the <stdlib.h> header file. */
#define HAVE_STDLIB_H 1

/* Define to 1 if you have the `strcasecmp' function. */
#define HAVE_STRCASECMP 1

/* Define to 1 if you have the `strdup' function. */
#define HAVE_STRDUP 1

/* Define to 1 if you have the `strftime' function. */
#define HAVE_STRFTIME 1

/* Define to 1 if you have the <strings.h> header file. */
#define HAVE_STRINGS_H 1

/* Define to 1 if you have the <string.h> header file. */
#define HAVE_STRING_H 1

/* Define to 1 if you have the `strncasecmp' function. */
#define HAVE_STRNCASECMP 1

/* Define to 1 if `tm_zone' is a member of `struct tm'. */
#define HAVE_STRUCT_TM_TM_ZONE 1

/* Define to 1 if you have the <sys/dir.h> header file, and it defines `DIR'.
   */
/* #undef HAVE_SYS_DIR_H */

/* Define to 1 if you have the <sys/file.h> header file. */
#define HAVE_SYS_FILE_H 1

/* Define to 1 if you have the <sys/ndir.h> header file, and it defines `DIR'.
   */
/* #undef HAVE_SYS_NDIR_H */

/* Define to 1 if you have the <sys/param.h> header file. */
#define HAVE_SYS_PARAM_H 1

/* Define to 1 if you have the <sys/rusage.h> header file. */
/* #undef HAVE_SYS_RUSAGE_H */

/* Define to 1 if you have the <sys/select.h> header file. */
#define HAVE_SYS_SELECT_H 1

/* Define to 1 if you have the <sys/socket.h> header file. */
#define HAVE_SYS_SOCKET_H 1

/* Define to 1 if you have the <sys/stat.h> header file. */
#define HAVE_SYS_STAT_H 1

/* Define to 1 if you have the <sys/time.h> header file. */
#define HAVE_SYS_TIME_H 1

/* Define to 1 if you have the <sys/types.h> header file. */
#define HAVE_SYS_TYPES_H 1

/* Define for Tcl that has Tcl_Free() (7.5p1 and later). */
#define HAVE_TCL_FREE 1

/* Define for Tcl that has Tcl_GetCurrentThread() (8.1a2 and later). */
#define HAVE_TCL_GETCURRENTTHREAD 1

/* Define for Tcl that has Tcl_GetThreadData() (8.1a2 and later). */
#define HAVE_TCL_GETTHREADDATA 1

/* Define for Tcl that has the Tcl_NotifierProcs struct member
   initNotifierProc (8.4 and later). */
#define HAVE_TCL_NOTIFIER_INIT 1

/* Define for Tcl that has Tcl_SetNotifier() (8.2b1 and later). */
#define HAVE_TCL_SETNOTIFIER 1

/* Define to 1 if your `struct tm' has `tm_zone'. Deprecated, use
   `HAVE_STRUCT_TM_TM_ZONE' instead. */
#define HAVE_TM_ZONE 1

/* Define to 1 if you don't have `tm_zone' but do have the external array
   `tzname'. */
/* #undef HAVE_TZNAME */

/* Define to 1 if you have the `uname' function. */
#define HAVE_UNAME 1

/* Define to 1 if you have the <unistd.h> header file. */
#define HAVE_UNISTD_H 1

/* Define to 1 if you have the `vprintf' function. */
#define HAVE_VPRINTF 1

/* Define to 1 if you have the `vsnprintf' function. */
#define HAVE_VSNPRINTF 1

/* Define to 1 if you have the <wchar.h> header file. */
#define HAVE_WCHAR_H 1

/* Define if modules will work on your system. */
#define MODULES_OK 1

/* Define if modules should be loaded using the dl*() functions. */
#define MOD_USE_DL 1

/* Define if modules should be loaded using the NS*() functions. */
/* #undef MOD_USE_DYLD */

/* Define if modules should be loaded using the ldr*() and *load() functions.
   */
/* #undef MOD_USE_LOADER */

/* Define if modules should be loaded using the rld_*() functions. */
/* #undef MOD_USE_RLD */

/* Define if modules should be loaded using the shl_*() functions. */
/* #undef MOD_USE_SHL */

/* Define to the address where bug reports for this package should be sent. */
#define PACKAGE_BUGREPORT "bugs@eggheads.org"

/* Define to the full name of this package. */
#define PACKAGE_NAME "Eggdrop"

/* Define to the full name and version of this package. */
#define PACKAGE_STRING "Eggdrop 1.6.21"

/* Define to the one symbol short name of this package. */
#define PACKAGE_TARNAME "eggdrop"

/* Define to the home page for this package. */
#define PACKAGE_URL ""

/* Define to the version of this package. */
#define PACKAGE_VERSION "1.6.21"

/* Define if running under QNX. */
/* #undef QNX_HACKS */

/* Define to the type of arg 1 for `select'. */
#define SELECT_TYPE_ARG1 int

/* Define to the type of args 2, 3 and 4 for `select'. */
#define SELECT_TYPE_ARG234 (fd_set *)

/* Define to the type of arg 5 for `select'. */
#define SELECT_TYPE_ARG5 (struct timeval *)

/* The size of `int', as computed by sizeof. */
#define SIZEOF_INT 4

/* The size of `long', as computed by sizeof. */
#define SIZEOF_LONG 8

/* The size of `short', as computed by sizeof. */
#define SIZEOF_SHORT 2

/* Define to 1 if you have the ANSI C header files. */
#define STDC_HEADERS 1

/* Define if running on OSF/1 platform. */
/* #undef STOP_UAC */

/* Define to 1 if you can safely include both <sys/time.h> and <time.h>. */
#define TIME_WITH_SYS_TIME 1

/* Define to 1 if your <sys/time.h> declares `struct tm'. */
/* #undef TM_IN_SYS_TIME */

/* Enable extensions on AIX 3, Interix.  */
#ifndef _ALL_SOURCE
# define _ALL_SOURCE 1
#endif
/* Enable GNU extensions on systems that have them.  */
#ifndef _GNU_SOURCE
# define _GNU_SOURCE 1
#endif
/* Enable threading extensions on Solaris.  */
#ifndef _POSIX_PTHREAD_SEMANTICS
# define _POSIX_PTHREAD_SEMANTICS 1
#endif
/* Enable extensions on HP NonStop.  */
#ifndef _TANDEM_SOURCE
# define _TANDEM_SOURCE 1
#endif
/* Enable general extensions on Solaris.  */
#ifndef __EXTENSIONS__
# define __EXTENSIONS__ 1
#endif


/* Define WORDS_BIGENDIAN to 1 if your processor stores words with the most
   significant byte first (like Motorola and SPARC, unlike Intel). */
#if defined AC_APPLE_UNIVERSAL_BUILD
# if defined __BIG_ENDIAN__
#  define WORDS_BIGENDIAN 1
# endif
#else
# ifndef WORDS_BIGENDIAN
/* #  undef WORDS_BIGENDIAN */
# endif
#endif

/* Define to 1 if on MINIX. */
/* #undef _MINIX */

/* Define to 2 if the system does not provide POSIX.1 features except with
   this defined. */
/* #undef _POSIX_1_SOURCE */

/* Define to 1 if you need to in order for `stat' and other things to work. */
/* #undef _POSIX_SOURCE */

/* Define to empty if `const' does not conform to ANSI C. */
/* #undef const */

/* Define to `__inline__' or `__inline' if that's what the C compiler
   calls it, or to nothing if 'inline' is not supported under any name.  */
#ifndef __cplusplus
/* #undef inline */
#endif

/* Define to the type of a signed integer type wide enough to hold a pointer,
   if such a type exists, and if the system does not define it. */
/* #undef intptr_t */

/* Define to `int' if <sys/types.h> does not define. */
/* #undef pid_t */

/* Define to `unsigned int' if <sys/types.h> does not define. */
/* #undef size_t */

/* Define to empty if the keyword `volatile' does not work. Warning: valid
   code using `volatile' can become incorrect without. Disable with care. */
/* #undef volatile */

#endif /* !_EGG_CONFIG_H */


src/compat/compat.h
/*
 * compat.h
 *   wrap-around header for all compability functions.
 *
 * $Id: compat.h,v 1.16 2011/07/09 15:07:48 thommey Exp $
 */
/*
 * Copyright (C) 2000 - 2011 Eggheads Development Team
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 */

#ifndef _EGG_COMPAT_COMPAT_H
#define _EGG_COMPAT_COMPAT_H

#include "inet_aton.h"
#include "snprintf.h"
#include "memset.h"
#include "memcpy.h"
#include "strcasecmp.h"
#include "strdup.h"
#include "strftime.h"

#endif /* !__EGG_COMPAT_COMPAT_H */


scripts/greetings.tcl
# ####
# INFO
# ####
#
# greetings.tcl v1.0 (Nov 01, 2007)
#
# ###########
# DESCRIPTION
# ###########
# 
# Channel Greetings for on join users with Random Welcome Messages, Set your channel via Partyline/DCC.
# Optional greeting style with your choice like; Greet users with Channel Message, Private Message or Notice.
# Ideal script for those who wish to set greeetings on specific channel(s) with optional style and random message.
#
# ########
# FEATURES
# ########
# 
# You can set your own greet messages.
# Set particular channel where you want to activate greeting system and also you can deactivate in the same manner.
# By default, this script greet user with on channel with any random message, even you can set greet style to wish user via MSG/Notice.
# you can change the settings as per your desire which is suitable for your channel.
#
# ############
# INSTALLATION
# ############
#
# * First unzip zipped file i.e. greetings.zip file 
# * Put greetings.tcl file in your eggdrop "/scripts" folder
# * Add a link at the bottom of your eggdrop's .CONF file
#
#   	   source scripts/greetings.tcl
#
# * Save your bot's configuration
# * RESTART your bot -OR- do .rehash from the partyline (DCC) to load the tcl.
#
# #######
# UPDATES
# #######
# 
# v0.0 - 01/08/2006 - Not Released, Beta Testing =)
# v1.0 - 01/11/2007 - Initial Release.
#
# #######
# CONTACT
# #######
#
# Any suggestions, comments, questions or bugs,
# feel free to email me at:
#
#			fyre_tcls@yahoo.com
#
# fyre @ #Eggdrop & #RO-TCL
# On Undernet IRC Network
#
# #########
# DOWNLOADS
# #########
#
# This script and other good TCL Scripts can be found on:
#
# * http://www.egghelp.org/
# * http://www.tclscript.com/
#
# -REGARDS-

########################
#- Channel Activation -#
########################

# Use .chanset to activate/deactivate channel greetings on the particular channel.
# Partyline/DCC :  n|n .chanset 
#       Example : .chanset #mychan1 +greetings
#                 .chanset #mychan2 -greetings

####################
#- Greeting Style -#
####################

# Set greeting style here, options are as follows.
# 0 - Channel Message
# 1 - Notice
# 2 - Private Message
set style "0"

###############
#- Greetings -#
###############

# Here you can set any join message for your channel.
set gmsg {
    "Welcome..."
    "Hello, how are you?"
    "Stay and Have Fun =)"
    "More msg..."
    "Here more..."
    "and it goes on..."
    "I Love You"
    "...............HAVE FUN!"}

########################################################
#- Don't edit below unless you know what you're doing -#
########################################################

bind join - * fyre_greet
setudef flag greetings

proc fyre_greet {nick uhost hand chan args} {
  global gmsg style botnick
  if {![channel get $chan greetings]} { return 0 }
  if {[isbotnick $nick]} {
    return 0
  }
  if {$style == "0"} { putserv "PRIVMSG $chan :$nick: [fyre_msg $args]" }
  if {$style == "1"} { putserv "NOTICE $nick :$nick: [fyre_msg $args]" }
  if {$style == "2"} { putserv "PRIVMSG $nick :$nick: [fyre_msg $args]" }
  return 0
}

proc fyre_msg {nick} {
   global gmsg
   set greetz [lindex $gmsg [rand [llength $gmsg]]]
   return "$greetz" 

}

#######################################################
putlog "LOADED: Greetings (greetings.tcl v1.0) by fyre"
#######################################################


src/patch.h
/*
 * patch.h
 *   header file for adding patches to Eggdrop's version string
 *
 * PLACE YOUR PATCHES HERE...
 *
 * HOW TO DO IT:
 *
 * replace a  *** PATCH GOES HERE *** line with a patch("patchname");
 * statement, leave the rest of the file alone, this allows better
 * overlapping patches.
 *
 * $Id: patch.h,v 1.1270 2011/10/25 16:45:08 thommey Exp $
 */
/*
 * Copyright (C) 1997 Robey Pointer
 * Copyright (C) 1999 - 2011 Eggheads Development Team
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License
 * as published by the Free Software Foundation; either version 2
 * of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
 */
#ifndef _EGG_PATCH_H
#define _EGG_PATCH_H
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* PATCH GOES HERE */
/*
 *
 *
 */
/* if that's not enough, maybe you should release a new version :) */
#endif /* _EGG_PATCH_H */
